robotoff/spellcheck/__init__.py

```diff
@@ -0,0 +1,70 @@
+from typing import Optional, Iterable
+
+from robotoff import settings
+from robotoff.utils.types import JSONType
+from robotoff.products import ProductDataset
+from robotoff.spellcheck.pipeline_spellchecker import PipelineSpellchecker
+
+
+class Spellchecker(PipelineSpellchecker):
+    def __init__(self, client, **es_kwargs):
+        super(Spellchecker, self).__init__()
+        self.es_kwargs = es_kwargs
+        self.add_spellchecker("patterns")
```

Comment:
Why not directly submitting a BaseSpellchecker instance? It seems more straightforward to me

Reply:
Another great addition would be to able to add/remove modules from the spellcheck directly here. A pipeline: List[str] parameter could be added, to be used this way:
```
pipeline = ["patterns", "percentages"]
Spellchecker(pipeline, client, **es_kwargs)
```

Reply:
As it is implemented at the moment :
- we have a BaseSpellchecker abstract class. It encapsulates an algorithm and has a method `correct (str -> str)`. Children are ElasticsearchSpellchecker, PatternsSpellchecker,...
- we have a PipelineSpellchecker class. It is used to encapsulate several BaseSpellcheckers in a pipeline. Most used function is `correct (str -> Item)`. A `add_spellchecker` method is used to stack the spellcheckers in the pipeline. I think that this class can benefit from a new method `from_pipeline(pipeline : List[str], **kwargs)`. The reason I didn't do it is that I didn't know how to gracefully takes arguments for the spellcheckers. For instance if we want to build a pipeline that is composed by 3 ElasticsearchSpellcheckers using different arguments (different indexes for instance). Calling several times `add_spellchecker` is more painful but more flexible. What's your opinion about that ?
- and finally we have a Spellchecker class that is a child of PipelineSpellchecker with a specific predefined pipeline. The objective is to have a default spellchecker that can be called from anywhere in robotoff without any overhead (the "best" configuration is saved in the spellchecker module and not outside). Maybe that's not what we want and we would prefer an API that is fully configurable. It easy to implement now but I wasn't sure we needed this use case.

Last thing, the `generate_insights` and `predict_insight` methods in Spellchecker are more related to the ProductDataset object and the generation of insights. It could be preferable to move them to the PipelineSpellchecker parent class.

Reply:
I might put this description in the documentation once we agree on the structure.